main/src/ca/uwaterloo/flix/language/phase/Monomorph.scala

```diff
@@ -124,683 +124,683 @@ object Monomorph extends Phase[TypedAst.Root, TypedAst.Root] {
   // TODO: Monomorph: We use exceptions here as a temporary stop-gap. We should consider to restructure and use Validation.
 
   /**
-    * Performs monomorphization of the given AST `root`.
+    * A function-local queue of pending (fresh symbol, function definition, and substitution)-triples.
+    *
+    * For example, if the queue contains the entry:
+    *
+    * -   (f$1, f, [a -> Int])
+    *
+    * it means that the function definition f should be specialized w.r.t. the map [a -> Int] under the fresh name f$1.
     */
-  def run(root: Root)(implicit flix: Flix): Validation[Root, CompilationMessage] = flix.phase("Monomorph") {
-
-    // TODO: Monomorph: Un-nest these.
-
-    /**
-      * A function-local queue of pending (fresh symbol, function definition, and substitution)-triples.
-      *
-      * For example, if the queue contains the entry:
-      *
-      * -   (f$1, f, [a -> Int])
-      *
-      * it means that the function definition f should be specialized w.r.t. the map [a -> Int] under the fresh name f$1.
-      */
-    val defQueue: mutable.Queue[(Symbol.DefnSym, Def, StrictSubstitution)] = mutable.Queue.empty
+  private type DefQueue = mutable.Queue[(Symbol.DefnSym, Def, StrictSubstitution)]
```

Comment:
Wait, but this is not safe. If two Flix compilers are running they will use the same queue!